packages/@aws-cdk/aws-cloudformation-codepipeline/lib/pipeline-action.ts

```diff
@@ -0,0 +1,318 @@
+     *
+     * Only present if configured by passing `outputFileName`.
+     */
+    public artifact?: codepipeline.Artifact;
+
+    constructor(parent: codepipeline.Stage, id: string, props: CloudFormationCommonProps, configuration?: any) {
+        super(parent, id, 'CloudFormation',  { minInputs: 0, maxInputs: 10, minOutputs: 0, maxOutputs: 1 }, {
+            ...configuration,
+            StackName: props.stackName,
+            OutputFileName: props.outputFileName,
+        });
+
+        if (props.outputFileName) {
+            this.artifact = this.addOutputArtifact(props.outputArtifactName || (this.parent!.name + this.name + 'Artifact'));
+        }
+    }
+}
+
+/**
+ * Properties for the ExecuteChangeSet action.
+ */
+export interface ExecuteChangeSetProps extends CloudFormationCommonProps {
+    /**
+     * Name of the change set to execute.
+     */
+    changeSetName: string;
+}
+
+/**
+ * CodePipeline action to execute a prepared change set.
+ */
+export class ExecuteChangeSet extends CloudFormationAction {
+    constructor(parent: codepipeline.Stage, id: string, props: ExecuteChangeSetProps) {
+        super(parent, id, props, {
+            ActionMode: 'CHANGE_SET_EXECUTE',
+            ChangeSetName: props.changeSetName,
+        });
+    }
+}
+
+/**
+ * Properties common to CloudFormation actions that stage deployments
+ */
+export interface CloudFormationDeploymentActionCommonProps extends CloudFormationCommonProps {
+    /**
+     * IAM role to assume when deploying changes.
+     *
+     * If not specified, a fresh role is created. The role is created with zero
+     * permissions unless `trustTemplate` is true, in which case the role will have
+     * full permissions.
+     *
+     * @default A fresh role with full or no permissions (depending on the value of `trustTemplate`).
+     */
+    role?: iam.Role;
+
+    /**
+     * Acknowledge certain changes made as part of deployment
+     *
+     * For stacks that contain certain resources, explicit acknowledgement that AWS CloudFormation
+     * might create or update those resources. For example, you must specify CAPABILITY_IAM if your
+     * stack template contains AWS Identity and Access Management (IAM) resources. For more
+     * information, see [Acknowledging IAM Resources in AWS CloudFormation Templates](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-iam-template.html#using-iam-capabilities).
+     *
+     * @default No capabitilities passed, unless `trustTemplate` is true
+     */
+    capabilities?: CloudFormationCapabilities[];
+
+    /**
+     * Whether the deployed templates are trusted.
+     *
+     * If `true`, the default `role` will have full permissions and the default
+     * `capabilities` will be set to `CloudFormationCapabilities.NamedIAM.`
+     *
+     * If `false`, you will have to set the appropriate IAM permissions and capabilities
+     * manually.
+     *
+     * @default false
+     */
+    trustTemplate?: boolean;
```

Comment:
I think something like `fullTrust` or `adminPrivileges` or `root`. Something that has to do with "admin"